docs/source/basic/faq.rst

```diff
@@ -56,3 +58,24 @@ Thunder is in alpha. There will be bugs, and many torch operations are not suppo
 If you need certain operations supported for your model, please let us know by creating an issue. We plan to get to all of them (with the exception of any :doc:`sharp edges <sharp_edges>`), but your issues help us prioritize which to do first.
 
 There are potentially any number of other problems which could arise. Some of the problems are known, some may not be. Check out the :doc:`sharp edges <sharp_edges>` page. If what you're seeing still doesn't make sense, let us know by creating an issue.
+
+
+=======================================
+5. Does Thunder support dynamic shapes?
+=======================================
+
+No, not at the moment. Meta functions operate on the exact shapes of the tensor proxies that pass through them. This is a limitation of the current implementation, and we plan to incorporate dynamic shapes in the future. We're looking for input on how to best incorporate dynamic shapes, so if you have any ideas, please let us know by creating an issue or reaching out.
+
+
+================================================================
+6. Does Thunder support (or plan to support) inplace operations?
+================================================================
+
+Not at the moment. Implementing inplace operations would require tracking which tensors in a trace have been modified by operations in our optimization passes, which currently we represent as purely functional. All deep learning compiler frameworks have to deal with the problem of tensor aliasing in some way. The way we've chosen for now is to pretend that the problem doesn't exist.
+
+The common solution is to represent programs in `SSA form <https://en.wikipedia.org/wiki/Static_single-assignment_form>`_, or do some form of SSA-inspired variable renaming, but this is a much less understandable representation than a list of symbols in a trace. At the same time, it would complicate optimization passes and require rewriting many of them to handle these aliasing rules.
+
+There also exists the problem that some backends, like Triton and nvfuser, or a potential Jax backend, are not actually good at handling inplace operations either. They, like Thunder, expect functional programs, and if we simply decide to forward these operations to the executor, not every executor would be able to implement every operator. For a simple torch operator like ``x.add_(y)``, this is unsatisfactory to us. We would ideally like to be able to convert instances of ``add_()`` to ``add()`` for executors. But every trace needs to be executable.
```

Comment:
I didn't understand how "But every trace needs to be executable" ties into the previous sentence. If we did the dummy thing of functionalizing `add_` into `add`, the trace would still be executable

Reply:
My thought was that the first trace, with the `add_` still in it, would not be. We could introduce some heuristic solution for this, for example marking traces as executable or intermediate, but we're still thinking things through.

Unrelated note, I think that an SSA-style tensor/variable renaming pass, without changing the program representation, plus marking traces as executable or not, is probably the solution that I favor. But we should all talk it over at some point, and validate with the nvfuser team that it's a good direction.

Reply:
Makes sense. Maybe this is clearer then

```suggestion
There also exists the problem that some backends, like Triton and nvfuser, or a potential Jax backend, are not actually good at handling inplace operations either. They, like Thunder, expect functional programs, and if we simply decide to forward these operations to the executor, not every executor would be able to implement every operator. For a simple torch operator like ``x.add_(y)``, this is unsatisfactory to us. We would ideally like to be able to convert instances of ``add_()`` to ``add()`` for executors. But currently every intermediate trace needs to be executable and `add_()` would not be a supported bound symbol before it is replaced later.
```